core/templates/dev/head/components/profileLink.js

```diff
@@ -0,0 +1,73 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directives for creating text and image links to a user's 
+ * profile page.
+ *
+ * @author raine@stanford.edu (Raine Hoover)
+ */
+
+oppia.directive('profileLinkText', [function() {
+  return {
+    restrict: 'E',
+    scope: {
+      username: '@'
```

Comment:
If you change '@' to '&' (here and below), you can do $scope.username() to get the username. This effectively makes the passed-in username read-only, which I think is what we want.